dist/gcc-compatible/Hacl_P256.c

```diff
@@ -213,2526 +285,1788 @@ static void sq(uint64_t *f, uint64_t *out)
+  qmul(tmp1, tmp1, x_11);
+  KRML_MAYBE_FOR5(i, (uint32_t)0U, (uint32_t)5U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_1111);
+  KRML_MAYBE_FOR5(i, (uint32_t)0U, (uint32_t)5U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_10101);
+  KRML_MAYBE_FOR4(i, (uint32_t)0U, (uint32_t)4U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_101);
+  KRML_MAYBE_FOR3(i, (uint32_t)0U, (uint32_t)3U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_101);
+  KRML_MAYBE_FOR3(i, (uint32_t)0U, (uint32_t)3U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_101);
+  KRML_MAYBE_FOR5(i, (uint32_t)0U, (uint32_t)5U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_111);
+  KRML_MAYBE_FOR9(i, (uint32_t)0U, (uint32_t)9U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_101111);
+  KRML_MAYBE_FOR6(i, (uint32_t)0U, (uint32_t)6U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_1111);
+  KRML_MAYBE_FOR2(i, (uint32_t)0U, (uint32_t)2U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, r);
+  KRML_MAYBE_FOR5(i, (uint32_t)0U, (uint32_t)5U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, r);
+  KRML_MAYBE_FOR6(i, (uint32_t)0U, (uint32_t)6U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_1111);
+  KRML_MAYBE_FOR5(i, (uint32_t)0U, (uint32_t)5U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_111);
+  KRML_MAYBE_FOR4(i, (uint32_t)0U, (uint32_t)4U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_111);
+  KRML_MAYBE_FOR5(i, (uint32_t)0U, (uint32_t)5U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_111);
+  KRML_MAYBE_FOR5(i, (uint32_t)0U, (uint32_t)5U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_101);
+  KRML_MAYBE_FOR3(i, (uint32_t)0U, (uint32_t)3U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_11);
+  KRML_MAYBE_FOR10(i, (uint32_t)0U, (uint32_t)10U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_101111);
+  KRML_MAYBE_FOR2(i, (uint32_t)0U, (uint32_t)2U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_11);
+  KRML_MAYBE_FOR5(i, (uint32_t)0U, (uint32_t)5U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_11);
+  KRML_MAYBE_FOR5(i, (uint32_t)0U, (uint32_t)5U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_11);
+  KRML_MAYBE_FOR3(i, (uint32_t)0U, (uint32_t)3U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, r);
+  KRML_MAYBE_FOR7(i, (uint32_t)0U, (uint32_t)7U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_10101);
+  KRML_MAYBE_FOR6(i, (uint32_t)0U, (uint32_t)6U, (uint32_t)1U, qsqr(tmp1, tmp1););
+  qmul(tmp1, tmp1, x_1111);
+  memcpy(x6, tmp1, (uint32_t)4U * sizeof (uint64_t));
+  memcpy(res, x6, (uint32_t)4U * sizeof (uint64_t));
+}
+
+static inline void qmul_mont(uint64_t *sinv, uint64_t *b, uint64_t *res)
+{
+  uint64_t tmp[4U] = { 0U };
+  from_qmont(tmp, b);
+  qmul(res, sinv, tmp);
+}
+
+static inline bool
+ecdsa_verify_msg_as_qelem(
```

Comment:
It looks like this code uses some vartime functions and some const_time functions. Is the intention that everything in here will eventually be vartime?